lib/workers/global/config/parse/additional-config-file.spec.ts

```diff
@@ -0,0 +1,288 @@
+import fs from 'node:fs';
+import fsExtra from 'fs-extra';
+import type { DirectoryResult } from 'tmp-promise';
+import { dir } from 'tmp-promise';
+import upath from 'upath';
+import { logger } from '../../../../logger';
+import customConfig from './__fixtures__/config';
+import * as file from './additional-config-file';
+
+describe('workers/global/config/parse/additional-config-file', () => {
+  const processExitSpy = vi.spyOn(process, 'exit');
+  const fsPathExistsSpy = vi.spyOn(fsExtra, 'pathExists');
+  const fsRemoveSpy = vi.spyOn(fsExtra, 'remove');
+
+  let tmp: DirectoryResult;
+
+  beforeAll(async () => {
+    tmp = await dir({ unsafeCleanup: true });
+  });
+
+  afterAll(async () => {
+    await tmp.cleanup();
+  });
+
+  describe('.getConfig()', () => {
+    it.each([
+      ['custom js config file', 'config.js'],
+      ['custom js config file', 'config.cjs'],
+      ['custom js config file', 'config.mjs'],
+      ['custom js config file exporting a Promise', 'config-promise.js'],
+      ['custom js config file exporting a function', 'config-function.js'],
+      // The next two are different syntactic ways of expressing the same thing
+      [
+        'custom js config file exporting a function returning a Promise',
+        'config-function-promise.js',
+      ],
+      [
+        'custom js config file exporting an async function',
+        'config-async-function.js',
+      ],
+      ['.renovaterc', '.renovaterc'],
+      ['JSON5 config file', 'config.json5'],
+      ['YAML config file', 'config.yaml'],
```

Comment:
Not sure if we have code style instruction regarding this, but usually we are using the other `it.each` style. 

https://github.com/renovatebot/renovate/blob/389554163b9acc83c32bcd94a794a78661b1547e/lib/util/array.spec.ts#L4